exchange/exchange.go

```diff
@@ -150,6 +144,8 @@ func (e *exchange) getAllBids(ctx context.Context, cleanRequests map[openrtb_ext
 			// Add in time reporting
 			elapsed := time.Since(start)
 			brw.adapterBids = bids
+			// validate bids ASAP, so we don't waste time on invalid bids.
+			err2 := brw.validateBids()
```

Comment:
Do we want to also count bid validation errors in the error metrics? If so, this could be something like:

```go
err = append(err, brw.validateBids()...)
```

And you'd no longer need the second loop further down.

Reply:
I do that below. I forgot that you could expand the list with `...`, so I did a loop and appended each one. I will have to fix that now.

Reply:
The `...` ellipsis [will handle the case of an empty array gracefully](https://play.golang.org/p/Qblz5-pxrLN), so this could just be:
```go
err = append(err, brw.validateBids()...)
```